README.md

```diff
@@ -135,43 +185,71 @@ Each release of rustls-ffi may introduce breaking changes to the ABI and so
 [SONAME](https://en.wikipedia.org/wiki/Soname).
 
-### Building a Dynamic Library
+# Example Applications
 
-Since building a useful dynamic library is more complex than building a static
-library, rustls-ffi uses [cargo-ci](https://github.com/lu-zero/cargo-c) in place
-of the `Makefile` system used for the static library.
+The rustls-ffi repo includes two example C programs for Linux, MacOS and Windows:
 
-This takes care of:
-* Generating the `rustls.h` header file.
-* Building a `.so` or `.dylib` file (depending on the OS).
-* Generating a [pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/) `.pc` file.
-* Installing the library and header files in the appropriate location.
+* [tests/client.c](tests/client.c) - a simple HTTPS client
+* [tests/server.c](tests/server.c) - a simple HTTPS server
+
+## Building the Examples
 
-If your operating system doesn't package `cargo-c` natively
-(see [package availability](https://github.com/lu-zero/cargo-c#availability)),
-you can install it with:
+To build the `client` and `server` C examples you will also need `cmake` (3.15+)
+and a C compiler (`gcc` and `clang` should both work, as well as MSVC on
+Windows).
 
-    cargo install cargo-c
+### Linux and MacOS
 
-To build a dynamic library in optimized mode:
+```
+# Configure your build directory
+cmake -S . -B build -DCMAKE_BUILD_TYPE=release
 
-    cargo capi build --release
+# Build the client/server examples
+cmake --build build
+```
 
-To install in `/usr/local/`:
+Use `-DCRYPTO_PROVIDER=ring` to select the cryptography provider for the
+examples explicitly.
 
-    sudo cargo capi install
+Use `-DCERT_COMPRESSION=on` to enable certificate compression.
 
-To build a static library in debug mode:
+Use `-DFIPS=on` to enable FIPS mode.
 
-    cargo capi build
+Use `-DDYN_LINK=on` to dynamically link Rustls to the test programs instead of
+statically linking (the default).
 
-To link against the resulting dynamic library, use `pkg-config` to populate your
-`LDLIBS` and `CFLAGS` as appropriate:
+Use `cmake --build build --target integration_test` to build and run the
+client/server integration tests.
 
-    LDLIBS="$(pkg-config --libs rustls)"
-    CFLAGS="$(pkg-config --cflags rustls)"
+Use `cmake -LH build` to see all options/help. Use `cmake --build build --target
+help` to view all targets.
+
+### Windows
+
+Note that Windows uses the traditional CMake "multi config" arrangement, using `--config` instead 
+of `-DCMAKE_BUILD_TYPE`:
+
+```
+# Configure your build directory
+cmake -S . -B build
+
+# Build the client/server examples
+cmake --build build --config Release
+```
+
+#### Examples
+
+Running the client/server examples in a debug build (ASAN enabled), with cert
+compression, the ring cryptography provider, and `clang` as the compiler on
+a Linux system in a `my-clang-build` build directory:
+
+```
+CC=clang CXX=clang cmake -S . -B my-clang-build -DCMAKE_BUILD_TYPE=Debug -DCERT_COMPRESSION=on
```

Comment:
Good catch! Thank you. 

> Presumably we need an explicit --debug instead of an explicit --release there.

I switched to using `"$<IF:$<CONFIG:Release>,--release,--debug>"` so that we're explicit in both cases and don't have to worry about upstream defaults. 

I was able to reproduce the problem you ran into and this fixed it for me locally.

Reply:
_(I also fixed the mistaken references to a `integration_test` target - it's `integration-test`)._